tests/test_image.py

```diff
@@ -353,6 +353,65 @@ def test_asnumpy_readonly():
         vw[0, 0, 0] = 123
 
 
+def test_save_overwrite(caplog):
+    """
+    Test that the overwrite flag behaves as expected.
+    - overwrite=True: Overwrites the existing file.
+    - overwrite=False: Raises an error if the file exists.
+    - overwrite=None: Renames the existing file and saves the new one.
+    """
+    im1 = Image(np.ones((1, 8, 8), dtype=np.float32))
+    im2 = Image(2 * np.ones((1, 8, 8), dtype=np.float32))
+    im3 = Image(3 * np.ones((1, 8, 8), dtype=np.float32))
+
+    # Create a tmp dir for this test output
+    with tempfile.TemporaryDirectory() as tmpdir_name:
+        # tmp filename
+        mrc_path = os.path.join(tmpdir_name, "og.mrc")
+        base, ext = os.path.splitext(mrc_path)
+
+        # Create and save the first image
+        im1.save(mrc_path, overwrite=True)
+
+        # Case 1: overwrite=True (should overwrite the existing file)
+        im2.save(mrc_path, overwrite=True)
+
+        # Load and check if im2 has overwritten im1
+        im2_loaded = Image.load(mrc_path)
+        np.testing.assert_allclose(im2.asnumpy(), im2_loaded.asnumpy())
+
+        # Case 2: overwrite=False (should raise an overwrite error)
+        with pytest.raises(
+            ValueError,
+            match="File '.*' already exists; set overwrite=True to overwrite it",
+        ):
+            im3.save(mrc_path, overwrite=False)
+
+        # Case 3: overwrite=None (should rename the existing file and save im3 with original filename)
+        with caplog.at_level(logging.INFO):
+            im3.save(mrc_path, overwrite=None)
+
+            # Check that the existing file was renamed and logged
+            assert f"Renaming {mrc_path}" in caplog.text
+
+            # Find the renamed file by checking the directory contents
+            renamed_file = None
+            for filename in os.listdir(tmpdir_name):
```

Comment:
Not a fan of this searching approach.

Assuming you made it to this line, it is possible to get the file form caplog but that is sort of gross too.  

It should possible to mock the `datetime` call to a known datetime, then you can check for file directly.  That sounds clean.

For example, here is some discussion:
https://stackoverflow.com/questions/4481954/trying-to-mock-datetime-date-today-but-not-working/25652721#25652721

The same approach could be used for other tests (test_volume etc).

Reply:
Mocked the `datetime` in both test_image and test_volume. Thanks!